src/NuGet.Clients/NuGet.CommandLine/Common/Solution.cs

```diff
@@ -111,8 +111,16 @@ private void LoadSolutionWithMsbuild4or12(Assembly msbuildAssembly, string solut
         // Load the solution file using the public class SolutionFile in msbuild 14
         private void LoadSolutionWithMsbuild14(Assembly msbuildAssembly, string solutionFileName)
         {
+            var isSolutionFilter = solutionFileName.EndsWith(".slnf", StringComparison.OrdinalIgnoreCase);
             var solutionFileType = msbuildAssembly.GetType("Microsoft.Build.Construction.SolutionFile");
             var parseMethod = solutionFileType.GetMethod("Parse", BindingFlags.Static | BindingFlags.Public);
+            var projectShouldBuildMethod = isSolutionFilter ? solutionFileType.GetMethod("ProjectShouldBuild", BindingFlags.NonPublic | BindingFlags.Instance)
```

Comment:
We should check for both public and non public methods? Just in case it becomes public.

Reply:
Good point.